tests/test_pbh.py

```diff
@@ -0,0 +1,795 @@
+                "flow_counter": "DISABLED"
+            }
+
+            attr_list = {
+                "gre_key": meta_dict["gre_key"],
+                "inner_ether_type": meta_dict["inner_ether_type"]
+            }
+
+            pbhlogger.info("Create PBH rule: {}".format(meta_dict["name"]))
+            dvs_pbh.create_pbh_rule(
+                table_name=meta_dict["table"],
+                rule_name=meta_dict["name"],
+                priority=meta_dict["priority"],
+                qualifiers=attr_list,
+                hash_name=meta_dict["hash"],
+                packet_action=meta_dict["packet_action"],
+                flow_counter=meta_dict["flow_counter"],
+            )
+            pbh_ref_count.incPbhRuleCount()
+            dvs_acl.verify_acl_rule_count(pbh_ref_count.getPbhRuleCount())
+
+            yield meta_dict
+
+        finally:
+            pbhlogger.info("Remove PBH rule: {}".format(meta_dict["name"]))
+            dvs_pbh.remove_pbh_rule(meta_dict["table"], meta_dict["name"])
+            pbh_ref_count.decPbhRuleCount()
+            dvs_acl.verify_acl_rule_count(pbh_ref_count.getPbhRuleCount())
+
+    @pytest.fixture
+    def pbh_vxlan(
+        self,
+        dvs_pbh,
+        dvs_acl,
+        pbh_ref_count,
+        pbh_table,
+        pbh_inner_v4
+    ):
+        try:
+            meta_dict = {
+                "table": pbh_table["name"],
+                "name": "vxlan",
+                "priority": "2",
+                "ip_protocol": "0x11/0xff",
+                "l4_dst_port": "0x12b5/0xffff",
+                "inner_ether_type": "0x0800/0xffff",
+                "hash": pbh_inner_v4["name"],
+                "packet_action": "SET_LAG_HASH",
+                "flow_counter": "ENABLED"
+            }
+
+            attr_list = {
+                "ip_protocol": meta_dict["ip_protocol"],
+                "l4_dst_port": meta_dict["l4_dst_port"],
+                "inner_ether_type": meta_dict["inner_ether_type"]
+            }
+
+            pbhlogger.info("Create PBH rule: {}".format(meta_dict["name"]))
+            dvs_pbh.create_pbh_rule(
+                table_name=meta_dict["table"],
+                rule_name=meta_dict["name"],
+                priority=meta_dict["priority"],
+                qualifiers=attr_list,
+                hash_name=meta_dict["hash"],
+                packet_action=meta_dict["packet_action"],
+                flow_counter=meta_dict["flow_counter"],
+            )
+            pbh_ref_count.incPbhRuleCount()
+            dvs_acl.verify_acl_rule_count(pbh_ref_count.getPbhRuleCount())
+
+            yield meta_dict
+
+        finally:
+            pbhlogger.info("Remove PBH rule: {}".format(meta_dict["name"]))
+            dvs_pbh.remove_pbh_rule(meta_dict["table"], meta_dict["name"])
+            pbh_ref_count.decPbhRuleCount()
+            dvs_acl.verify_acl_rule_count(pbh_ref_count.getPbhRuleCount())
+
+    def test_PbhNvgreVxlanConfiguration(self, testlog, pbh_nvgre, pbh_vxlan):
```

Comment:
Suggestion: can we have a seperate test for vxlan, nvgre?
If you create a single fixture pbh_cfg parametrized with nvgre and vxlan it would also resolve this comment as it will show up as two test cases in pytest reports